src/Microsoft.ML.StandardTrainers/Standard/ModelStatistics.cs

```diff
@@ -16,249 +16,345 @@
+            }
+            else
+            {
+                int[] weightsIndices = ctx.Reader.ReadIntArray(weightsLength);
+                _weights = new VBuffer<float>(weightsLength, weightsLength, stdErrorValues, weightsIndices);
+            }
         }
 
-        private void SaveCore(ModelSaveContext ctx)
+        private protected override void SaveCore(ModelSaveContext ctx)
         {
             // *** Binary Format ***
-            // int: count of parameters
-            // long: count of training examples
-            // float: deviance
-            // float: null deviance
-            // bool: whether standard error is included
+            // <base>
             // (Conditional) float[_paramCount]: values of std errors of coefficients
             // (Conditional) int: length of std errors of coefficients
             // (Conditional) int[_paramCount]: indices of std errors of coefficients
 
-            _env.Assert(_paramCount > 0);
-            ctx.Writer.Write(_paramCount);
-
-            _env.Assert(_trainingExampleCount > 0);
-            ctx.Writer.Write(_trainingExampleCount);
+            base.SaveCore(ctx);
 
-            ctx.Writer.Write(_deviance);
-            ctx.Writer.Write(_nullDeviance);
-
-            bool hasStdErrors = _coeffStdError.HasValue;
-            ctx.Writer.Write(hasStdErrors);
-            if (!hasStdErrors)
-                return;
-
-            var coeffStdErrorValues = _coeffStdError.Value.GetValues();
-            _env.Assert(coeffStdErrorValues.Length == _paramCount);
+            var coeffStdErrorValues = _coeffStdError.GetValues();
+            Env.Assert(coeffStdErrorValues.Length == ParametersCount);
             ctx.Writer.WriteSinglesNoCount(coeffStdErrorValues);
-            ctx.Writer.Write(_coeffStdError.Value.Length);
-            if (_coeffStdError.Value.IsDense)
-                return;
-
-            ctx.Writer.WriteIntsNoCount(_coeffStdError.Value.GetIndices());
+            ctx.Writer.Write(_coeffStdError.Length);
+            if (!_coeffStdError.IsDense)
+                ctx.Writer.WriteIntsNoCount(_coeffStdError.GetIndices());
+
+            //save the bias
+            ctx.Writer.Write(_bias);
+
+            //save the weights
+            ctx.Writer.WriteBoolByte(_weights.IsDense);
+            ctx.Writer.Write(_weights.Length);
+            ctx.Writer.WriteSinglesNoCount(_weights.GetValues());
+            if (!_weights.IsDense)
+                ctx.Writer.WriteIntsNoCount(_coeffStdError.GetIndices());
         }
 
         /// <summary>
-        /// Computes the standart deviation, Z-Score and p-Value.
+        /// Computes the standart deviation, Z-Score and p-Value for the value being passed as the bias.
         /// </summary>
-        public static bool TryGetBiasStatistics(LinearModelStatistics stats, float bias, out float stdError, out float zScore, out float pValue)
+        public CoefficientStatistics GetBiasStatisticsForValue(float bias)
         {
```

Comment:
Is it necessary to users? Could we internalize it?

Reply:
it has current external usage, should keep it. I'll add a functional test too. 

---
In reply to: [266657218](https://github.com/dotnet/machinelearning/pull/2048#discussion_r266657218) [](ancestors = 266657218)